openapi-to-har.js

```diff
@@ -20,6 +20,26 @@
  */
 const OpenAPISampler = require('openapi-sampler');
 
+/**
+ * Looks at the first entry in examples and if it has a value of type string, it returns that.
+ *
+ * If the object is empty, or the first entry has a value whose type is not a string, then the
+ * defaultValue is returned.
+ *
+ * @param {Object} examples      An OpenAPI Examples object
+ * @param {string} defaultValue  The default value to use for the parameter if no suitable value found
+ * @return {string}              A suitable value from examples object, else defaultValue
+ */
+const getParameterValueFromExamples = function (examples, defaultValue) {
+  if (examples && typeof examples === 'object') {
+    const values = Object.values(examples);
+    if (values.length > 0 && typeof values[0].value === 'string') {
```

Comment:
what if first entry is a reference object. Is that already "inlined" at this point? Will need to test it.

Reply:
So if the first entry is a reference, then the default value is returned. Would be nice if the reference was followed.

Reply:
Also, if the value is a `number` or `boolean` then this check seems too restrictive. 

So here's my idea (which tries to stick close to the original author's idea for grabbing `example`)

1. Let's call `values[0].value` the `potentialCandidateValue`
2. if `potentialCandidateValue` is of type `string`, `boolean`, `number` or it's `null` then return them (as a string)
3. If `potentialCandidateValue` is of type `array` then just return the first element of that array (as a string). This is done because there are different ways of modeling arrays as query parameters. So simpler to just default to a single value
4. Else just return `defaultValue`

Reply:
Something like this:

```js
const getParameterValueFromExamples = function (examples, defaultValue) {
  if (examples && typeof examples === 'object') {
    const values = Object.values(examples);

    if (values.length > 0 && typeof values[0].value !== 'undefined') {
      const potentialValue = values[0].value;
      const potentialValueType = typeof potentialValue;

      switch (potentialValueType) {
        case 'boolean':
        case 'string':
        case 'number':
        case 'bigint':
          return potentialValue + '';
        case 'object':
          if (potentialValue === null) {
            return 'null';
          }
          break;
        case 'array':
          // just return the first entry since there are multiple ways to model
          // an array as a query parameter.
          if (potentialValue.length > 0) {
            return potentialValue[0].value + '';
          }
          break;
      }
    }
  }
  return defaultValue;
};
```

Reply:
A simpler approach that is really more in line with how `example` is parsed

1. If it's an array grab the first value
2. For everything else just return `values[0].value + ''` (convert it to a string).

This would then work just like `example` except in the case that we have an array in which case we just return the first value of the array.

Reply:
So this:

```js
const getParameterValueFromExamples = function (examples, defaultValue) {
  if (examples && typeof examples === 'object') {
    const values = Object.values(examples);

    if (values.length > 0 && typeof values[0].value !== 'undefined') {
      const potentialValue = values[0].value;

      if (Array.isArray(potentialValue)) {
        // just return the first entry since there are multiple ways to model
        // an array as a query parameter.
        if (potentialValue.length > 0) {
          return potentialValue[0] + '';
        }
      }

      return potentialValue + '';
    }
  }
  return defaultValue;
};

```

Reply:
> I just realized that array is not an output of typeof. So I'll need to check if something is an array in the object case.

One of the more annoying nuances of JavaScript. I suggest [`Array.isArray()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) for that particular check, though that isn't especially helpful in a switch block. Since most of the checks just convert to a string anyway I suggest removing the switch and going with something like:

```js
const getParameterValueFromExamples = function (examples, defaultValue) {
  if (examples && typeof examples === 'object') {
    const values = Object.values(examples);

    if (values.length > 0 && typeof values[0].value !== 'undefined') {
      const potentialValue = values[0].value;

      if (potentialValue === null) {
        return null;
      }

      if (Array.isArray(potentialValue)) {
        // make potentialValue the first entry since there are multiple ways to
        // model an array as a query parameter.
        potentialValue = potentialValue[0];
      }

      return potentialValue ? `${potentialValue}` : defaultValue;
  }
  return defaultValue;
};
```

Reply:
So my solution breaks tests as there is already some support for arrays which I didn't notice previously.